pkg/system_api.go

```diff
@@ -109,7 +109,7 @@ func (components *Components) UnmarshalJSON(b []byte) error {
 	return nil
 }
 
-func (components Components) CreateTreeStrcuture() Components {
+func (components Components) CreateTreeStrcuture(queryID, status string) Components {
```

Comment:
I think just pass through the entire params object, also there is a typo

Reply:
We can't pass the entire object at that creates the import cycle

Reply:
Ok - can we fix the typo ? CreateTreeStructure

Reply:
Pushed